### PR TITLE
doc: Include ruleset files in documentation

### LINKED ISSUE #527: Add new ruleset files with common rules.
Now that we can use Rulesets and export them as files, we should define the required, export them, and upload it to the user guide so users can download them and import them into their projects.

### PR DESCRIPTION
**Pull request by AlejandroFernandezLuces**
Closes #527

### CODE REVIEW

File: doc/styles/config/vocabularies/ANSYS/accept.txt [hunk 1]
```diff
@@ -91,7 +91,8 @@ reStructuredText
 REST
 RPC
 RST
-rulesets
+(?i)rulesets
+(?i)ruleset
```
Comment (SMoraisAnsys) [inline doc/styles/config/vocabularies/ANSYS/accept.txt:95]:
```suggestion
[Rr]ulesets
[Rr]uleset
```
Otherwise rUlEsEt would be a valid  name

**Review by SMoraisAnsys — APPROVED** [review]
LGTM, I left a minor comment

File: doc/styles/config/vocabularies/ANSYS/accept.txt (hunk 1 above)
Comment (AlejandroFernandezLuces) [reply doc/styles/config/vocabularies/ANSYS/accept.txt:95]: Makes sense, thanks 🙂

File: doc/source/_rulesets/main.json [hunk 2]
```diff
@@ -0,0 +1,44 @@
+{
+  "id": 2045316,
+  "name": "Main branch protection",
+  "target": "branch",
+  "source_type": "Repository",
+  "source": "ansys-internal/pymechanical-meshworkflows",
```
Comment (jorgepiloto) [inline doc/source/_rulesets/main.json:6]: I do not think this is important but we could use a repo that is public.

**Review by jorgepiloto — COMMENTED** [review]
Thanks for adding this, @AlejandroFernandezLuces. Just left a comment.

File: doc/source/_rulesets/main.json (hunk 2 above)
Comment (RobPasMue) [reply doc/source/_rulesets/main.json:6]: Please do... @AlejandroFernandezLuces remove this

**Comment by greschd** [comment]
@AlejandroFernandezLuces having used this, I have two questions / comments:
- shouldn't the "branch naming" protection apply to all branches, not just `main`?
- it appears `docs/` is missing in the list of allowed branch name patterns.

**Comment by AlejandroFernandezLuces** [comment]
> @AlejandroFernandezLuces having used this, I have two questions / comments:
> 
> * shouldn't the "branch naming" protection apply to all branches, not just `main`?
> * it appears `docs/` is missing in the list of allowed branch name patterns.

Agreed, thanks for raising those points. I'll make the fixes in a follow up PR.

**Pull request merged**